### DISCUSSION TITLE
Overlay, progress indicator and drop down: Prototyping new components

### DISCUSSION THREAD

**Discussion by khawkins98** [visual-framework/vf-core#1643] [Ideas]
The EBI Training team is working on a range of new features. @cindyebi @nikiforosk @nitinja @prakashebi and Mahfouz discussed them today — we agreed on some basics to ease future development of these in the VF (and to minimise breaking changes for Training):

- Namespacing:
    - vf-overlay
    - vf-progress-indicator
    - vf-dropdown (perhaps to be moved to vf-navigation--dropdown or vf-menu--dropdown, but that can be done at a later date)
- CSS: 
    - should follow [the VF's namespacing convention and other guidance](https://stable.visual-framework.dev/guidance/css-and-sass/index.html?q=sass)
    - example CSS: [vf-tabs](https://github.com/visual-framework/vf-core/blob/develop/components/vf-tabs/vf-tabs.scss)
- JavaScript
    - should follow [the VF's JS conventions](https://stable.visual-framework.dev/guidance/javascript/)
    - example JS: [vf-tabs.js](https://github.com/visual-framework/vf-core/blob/develop/components/vf-tabs/vf-tabs.js)

Once the components are live on training, we should be able to use that code as the basis for future development, testing and documentation.

**Comment by cindyebi** [comment]
An example of the _vf-dropdown navigation:_
![Screenshot 2021-08-25 at 15 03 58](https://user-images.githubusercontent.com/75834903/130805509-7a886542-5381-4fb7-bbec-5c51d8e7c334.png)

An example of a _vf-progress-indicator_:
![Screenshot 2021-08-25 at 15 04 08](https://user-images.githubusercontent.com/75834903/130805510-d7844ea3-6074-4793-a4f8-ad2a31d54104.png)

**Comment by khawkins98** [comment]
Further conversation is happening at:

-  Discussion: navigation drop-down menus #1299 
-  Progress Indicator #1648
